storagemarket/testnodes/testnodes.go

```diff
@@ -103,20 +108,28 @@ func (n *FakeCommonNode) GetChainHead(ctx context.Context) (shared.TipSetToken,
 }
 
 // AddFunds adds funds to the given actor in the storage market state
-func (n *FakeCommonNode) AddFunds(ctx context.Context, addr address.Address, amount abi.TokenAmount) error {
+func (n *FakeCommonNode) AddFunds(ctx context.Context, addr address.Address, amount abi.TokenAmount) (cid.Cid, error) {
 	n.SMState.AddFunds(addr, amount)
-	return nil
+	return n.AddFundsCid, nil
 }
 
 // EnsureFunds adds funds to the given actor in the storage market state to insure it has at least the given amount
-func (n *FakeCommonNode) EnsureFunds(ctx context.Context, addr, wallet address.Address, amount abi.TokenAmount, tok shared.TipSetToken) error {
+func (n *FakeCommonNode) EnsureFunds(ctx context.Context, addr, wallet address.Address, amount abi.TokenAmount, tok shared.TipSetToken) (cid.Cid, error) {
 	if n.EnsureFundsError == nil {
 		balance := n.SMState.Balance(addr)
 		if balance.Available.LessThan(amount) {
-			n.SMState.AddFunds(addr, big.Sub(amount, balance.Available))
+			return n.AddFunds(ctx, addr, big.Sub(amount, balance.Available))
 		}
 	}
-	return n.EnsureFundsError
+
+	return cid.Undef, n.EnsureFundsError
+}
+
+func (n *FakeCommonNode) WaitForMessage(mcid cid.Cid, confidence int64, onCompletion func(exitcode.ExitCode, []byte) error) error {
+	if n.BlockOnMessageWait {
+		time.Sleep(5 * time.Second)
```

Comment:
not a huge fan of time.sleeps in testing... does this serve a specific purpose?
btw, after we get https://github.com/filecoin-project/go-fil-markets/issues/77 done I hope we can remove the time.sleep calls entirely.